frontend/src/components/common/AddAdminModal.tsx

```diff
@@ -0,0 +1,189 @@
+      return;
+    }
+    console.log("creating admin with data");
+    console.log(
+      `first name: ${firstName}\nlast name: ${lastName}\nemail: ${email}`,
+    );
+    onModalClose();
+  };
+
+  return (
+    <>
+      <Button
+        rightIcon={<PlusOutlineIcon />}
+        variant="primary"
+        my={2}
+        onClick={onOpen}
+      >
+        Add Admin
+      </Button>
+      <Modal isOpen={isOpen} onClose={onModalClose} size="6xl" isCentered>
+        <ModalOverlay />
+        <ModalContent p={2}>
+          <ModalHeader>
+            <Text textStyle="subtitle1" color="grey.400">
+              Add Admin
+            </Text>
+          </ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            <FormControl isRequired>
+              <AddAdminFormLabel text="Name of Admin" />
+              <HStack direction="row" mt={6}>
+                <AddAdminTextInput
+                  validationType="text"
+                  placeholder="First Name"
+                  handleChange={setFirstName}
+                />
+                <AddAdminTextInput
+                  validationType="text"
+                  placeholder="Last Name"
+                  handleChange={setLastName}
+                />
+              </HStack>
+            </FormControl>
+            <FormControl isRequired as="fieldset" mt={8}>
+              <AddAdminFormLabel text="Does the user already have a Jump Math email address?" />
+              <RadioGroup
+                onChange={(val) => setHasJumpMathEmail(val === "yes")}
+                mt={5}
+              >
+                <HStack spacing="24px">
+                  <Radio value="no" mt={2} outlineColor="grey.300" size="lg">
+                    No
+                  </Radio>
+                  <Radio value="yes" outlineColor="grey.300" size="lg">
+                    Yes
+                  </Radio>
+                </HStack>
+              </RadioGroup>
+            </FormControl>
+            {hasJumpMathEmail !== null && (
+              <>
+                <FormControl isRequired mt={6}>
+                  <AddAdminFormLabel
+                    text={`Please enter their ${
+                      hasJumpMathEmail ? "Jump Math" : ""
+                    } email address`}
+                  />
+                  <AddAdminTextInput
+                    validationType="email"
+                    placeholder="Email Address"
+                    handleChange={setEmail}
+                  />
+                </FormControl>
+                <FormControl isRequired mt={6}>
+                  <AddAdminFormLabel text="Confirm email address" />
+                  <AddAdminTextInput
+                    validationType="email"
+                    placeholder="Email Address"
+                    handleChange={setConfirmEmail}
+                  />
+                </FormControl>
+              </>
+            )}
+          </ModalBody>
+          <ModalFooter>
```

Comment:
maybe we can separate this out into a separate component? We can probably use this at another point in time anyway.